Make the high-level ACP API unallow, not deny

Vinnl opened the pull request:
This changes the high-level ACP API to, when the value `false` is
passed for an access mode, remove allow statements for that access
mode, rather than adding a deny statement.

There are two reasons for this:
1. This aligns it with the WAC behaviour.
2. This allows the developer to actually undo previously set
   access without unintended consequences. For example, removing
   access for an agent shouldn't prevent their access if a group
   they're in still has access.

I'm a bit worried that it was a little bit _too_ straightforward to get all the tests succeeding, and the change itself was smaller than I expected as well, but since we can't test it against an actual server implementation yet, I can only cross my fingers and hope that it works as intended :)

- [x] I've added a unit test to test for potential regressions of this bug.
- [x] The changelog has been updated, if applicable.
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

Vinnl commented:
Ah, forgot that the WAC API also needed changes to ensure that the `getAccess` functions did not return undefined for access modes: dcd41d26 (I'll squash that into the other commit before merging.)

NSeydoux reviewed (approved):
As part of the PR, I think all the `@returns` annotations in the WAC high-level API could be removed, provided the description of the `Access` type is part of the API documentation. Right now, the `@returns` still mention `true`, `false` or `undefined` for returned access modes, and there are no `@returns` annotations on the ACP high-level functions. Avoiding to have duplicate information between the `@returns` and the `Access` description seems to be the best approach.

Vinnl commented:
Ah nice catch. I updated the descriptions to state _what_ it returns (i.e. access for the given actor), but not how that's structured. That also avoids duplicate information :) And I added `@returns` annotations to the ACP API while I was at it: 26825f8e
